### PR TITLE
chore(tokengen): add server binary

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
- Refactored `tokengen/main.rs` into `main.rs` and `lib.rs` for better code organization and reusability.
- Introduced a new binary target that runs as a server to generate tokens.

This server enhances the development experience by:
1. Simplifying token generation for web developers, eliminating the manual process of using the CLI and copying tokens to access the gateway.
2. Paving the way for future integration with IronRDP web demos, enabling smoother token handling.

### CODE REVIEW

**Review by CBenoit — COMMENTED** [review]
The idea would be to start this server and let the demo web client request tokens on its own (without authentication)?

Can I suggest you to keep both in a single binary? I think it would not be too big, and we already have a workflow for tokengen itself.

Nest the CLI into two new subcommands:
- `tokengen sign`
- `tokengen server`

I also suggest that you re-use the route we already have in the Devolutions Gateway for the webapp:
https://github.com/Devolutions/devolutions-gateway/blob/eac37e81e3006d747543e331a05137c9281f1109/devolutions-gateway/src/api/webapp.rs#L222-L453

I understand we only need to support session tokens to allow the demo web clients to generate session tokens. For instance, we don’t need to support JRL tokens.

File: tools/tokengen/src/main.rs [hunk 1]
```diff
@@ -1,215 +1,85 @@
+// main.rs
+
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply tools/tokengen/src/main.rs]: updated

File: tools/tokengen/src/bin/server/main.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    "{}=debug,tower_http=debug,axum::rejection=trace",
+                    env!("CARGO_CRATE_NAME")
+                )
+                .into()
+            }),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    // Read the provisioner key path from the configuration
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply tools/tokengen/src/bin/server/main.rs]: updated

File: tools/tokengen/src/server/server_impl.rs [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+}
+
+#[derive(Deserialize)]
+pub(crate) struct CommonRequest {
+    #[serde(default)]
+    validity_duration: Option<String>,
+    #[serde(default)]
+    kid: Option<String>,
+    #[serde(default)]
+    delegation_key_path: Option<String>,
```
Comment (CBenoit) [inline tools/tokengen/src/server/server_impl.rs:62]: I think this should not be part of the request. Why not proceed like for `provisioner_key_path`?

Comment (CBenoit) [inline tools/tokengen/src/server/server_impl.rs:62]: I think this should not be part of the request. Why not proceed like for `provisioner_key_path`?

File: tools/tokengen/src/server/server_impl.rs [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        provisioner_key_path
+    };
+
+    Ok(Arc::new(provisioner_key_path))
+}
+
+#[derive(Deserialize)]
+pub(crate) struct CommonRequest {
+    #[serde(default)]
+    validity_duration: Option<String>,
```
Comment (CBenoit) [inline tools/tokengen/src/server/server_impl.rs:58]: naming: `lifetime: Option<u64>` (the lifetime in seconds)

**Review by CBenoit — COMMENTED** [review]
> Would it be reasonable to keep the API as it is for now? The tokengen app serves a different purpose (mostly for testing), and occasionally we might need more granular control over the token content. Let me know your thoughts on this approach.

Fair enough.
Let me suggest you a few modifications on what you already submitted then.

File: tools/tokengen/src/server/server_impl.rs (hunk 3 above)
Comment (irvingoujAtDevolution) [reply tools/tokengen/src/server/server_impl.rs:62]: updated

File: tools/tokengen/src/server/server_impl.rs (hunk 4 above)
Comment (irvingoujAtDevolution) [reply tools/tokengen/src/server/server_impl.rs]: updated

**Review by CBenoit — APPROVED** [review]
LGTM

**Pull request merged**